accounts.ts

```diff
@@ -0,0 +1,121 @@
+import { Client, Account, AccountCorrent, AccountSavings } from "./interfaces";
+
+// Base para a  conta
+abstract class AccountBase implements Account {
```

Comment:
Exelente uso da classe abstrata para coisas em comum. 

De ponto de melhoria, conforme o sistema for crescendo, talves será necessário isolar os  métodos. 
Tenta sempre pensar no conceito de responsabilidade.